src/Microsoft.AspNet.Mvc.ViewFeatures/Rendering/Html/TagBuilder.cs

```diff
@@ -195,56 +177,44 @@ public void MergeAttribute(string key, string value, bool replaceExisting)
 
         public void SetInnerText(string innerText)
         {
-            InnerHtml = _htmlEncoder.HtmlEncode(innerText);
-        }
-
-        public HtmlString ToHtmlString(TagRenderMode renderMode)
-        {
-            return new HtmlString(ToString(renderMode));
+            InnerHtml = new StringHtmlContent(innerText);
         }
```

Comment:
Don't we need it to indicate an entry in `BufferedHtmlContent` (or similar) needs to be encoded when `WriteTo()` is called?